spring-xd-dirt/src/main/java/org/springframework/xd/dirt/job/dsl/JobSpecification.java

```diff
@@ -0,0 +1,557 @@
+	}
+
+	/**
+	 * Basic visitor that simply collects up any inlined job definitions.
+	 */
+	static class JobDefinitionLocator extends JobSpecificationVisitor<Object> {
+
+		List<JobDefinition> jobDefinitions = new ArrayList<JobDefinition>();
+
+		public List<JobDefinition> getJobDefinitions() {
+			return jobDefinitions;
+		}
+
+		@Override
+		public Object walk(Object context, Flow sjs) {
+			for (JobNode jobNode : sjs.getSeries()) {
+				walk(context, jobNode);
+			}
+			return context;
+		}
+
+		@Override
+		public Object walk(Object context, JobDefinition jd) {
+			jobDefinitions.add(jd);
+			return context;
+		}
+
+		@Override
+		public Object walk(Object context, JobReference jr) {
+			return context;
+		}
+
+		@Override
+		public Object walk(Object context, Split pjs) {
+			for (JobNode jobNode : pjs.getSeries()) {
+				walk(context, jobNode);
+			}
+			return context;
+		}
+
+	}
+
+	/**
+	 * Visitor that produces an XML representation of the Job specification.
+	 */
+	static class XMLGeneratorVisitor extends JobSpecificationVisitor<Element[]> {
+
+		/**
+		 * containing document that can be used for element creation
+		 */
+		private Document doc;
+
+		/**
+		 * Where to append elements created during the visit
+		 */
+		private Element batchJobElement;
+
+		/**
+		 * Should the XML output be readable or compressed onto one line.
+		 */
+		private boolean prettyPrint;
+
+		/**
+		 * A stack that tracks the element that should have new children attached to it.
+		 * Initially populated with batchJobElement from above.
+		 */
+		private Stack<Element> currentElement = new Stack<>();
+
+		/**
+		 * Counter for the numeric suffix to attach to generated split id attributes.
+		 */
+		private int splitId = 1;
+
+		private List<String> jobRunnerIds = new ArrayList<>();
+
+		private String xmlString;
+
+		private String batchJobId;
+
+		XMLGeneratorVisitor(String batchJobId, boolean prettyPrint) {
+			this.batchJobId = batchJobId;
+			this.prettyPrint = prettyPrint;
+		}
+
+		public String getXmlString() {
+			return xmlString;
+		}
+
+		@Override
+		public void preJobSpecWalk() {
+			try {
+				DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+				DocumentBuilder db = dbf.newDocumentBuilder();
+				DOMImplementation domImplementation = db.getDOMImplementation();
+
+				// Generate:
+				// <beans xmlns="http://www.springframework.org/schema/beans"
+				//		  xmlns:batch="http://www.springframework.org/schema/batch"
+				//		  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+				//		  xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd
+				//		    http://www.springframework.org/schema/batch http://www.springframework.org/schema/batch/spring-batch.xsd">
+				this.doc = domImplementation.createDocument("http://www.springframework.org/schema/beans", "beans",
+						null);
+				doc.createElementNS("http://www.springframework.org/schema/batch", "batch");
```

Comment:
also had to use xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" else I received SAX exception